### ISSUE TITLE
Document Twisted copyright policy

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11540]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)| @puzzlement reported|
|-|-|
|Trac ID|trac#1262|
|Type|defect|
|Created|2005-10-09 08:33:08Z|
|Branch|https://github.com/twisted/twisted/tree/copyright-policy-1262|

<details><summary>Searchable metadata</summary>

```
trac-id__1262 1262
type__defect defect
reporter__hypatia hypatia
priority__high high
milestone__ 
branch__branches_copyright_policy_1262 branches/copyright-policy-1262
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1128846788000000 1128846788000000
changetime__1220912423000000 1220912423000000
version__ 
owner__ 
cc__glyph cc__radix cc__robertc cc__thijs cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Sorry about this can of worms, folks.

Coming out of yesterday's discussion on #twisted: lifeless wishes to retain
copyright of one of his own files when he contributes it to Twisted under the
MIT licence. As I understand it, this is compatible with Glyph's last statement
on the subject at
http://twistedmatrix.com/pipermail/twisted-python/2005-August/011255.html

lifeless also questioned how we do this: specifically, do we strip other
people's copyright statements out of individual MIT files that originate
elsewhere and which are contributed to Twisted. radix was unsure. It seems we
shouldn't take these statements and move them to the LICENSE file, as best I can
tell. Any ideas?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Oh and does anyone have a good sense of a shorthand for referring to our
copyright. Is "Copyright 200x-2005 Twisted Matrix Laboratories except where
noted in individual files" or anything like it a meaningful or correct statement
any more?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
I think that technically speaking we're not supposed to remove copyright
statements from individual files, but they also become gradually invalid
as more contributors add significant portions to particular files.
Since MIT license says not to remove the notice, I don't think we can
make a policy of removing it.

In this case, we'd ask lifeless to remove the copyright notices if he is
comfortable with doing that, and if not, leave them there.  It's just an
aesthetic preference of mine that license crap not pollute the top of
every single file in the system.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement set owner to edsuom|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Are there any wiki pages for the current Twisted copyright policy? Should we create a .xhtml doc? It can be a few lines, we'd at least need something (and this ticket indicates there's nothing atm).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

(In [24697]) Branching to 'copyright-policy-1262'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Added a copyright header section to the coding standard doc in r24698, putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Perhaps this should also discuss what to do in the case described in the first comment on this ticket.  I think the resolution is that we leave it, but once we start editing the file, we also add the Twisted copyright.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [exarkun](#note_9):
> Perhaps this should also discuss what to do in the case described in the first comment on this ticket.  I think the resolution is that we leave it, but once we start editing the file, we also add the Twisted copyright.

I added a note for this in r24732 and rewrote the XXX part a little. Up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24747]) Merge copyright-policy-1262: Added a copyright header section to the coding standard doc.

Author: thijs
Reviewer: exarkun
Fixes: [#11540](https://github.com/twisted/twisted/issues/11540)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
